src/lib.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::kbc_modules::{KbcCheckInfo, KbcInstance, KbcModuleList};
+use anyhow::*;
+use std::collections::HashMap;
+
+mod kbc_modules;
+
+/// Attestation Agent (AA for short) is a rust library crate for attestation procedure
+/// in confidential containers. It provides kinds of service APIs that need to make
+/// requests to the Relying Party (Key Broker Service) in Confidential Containers,
+/// and establishes an attestation and connection between the corresponding Key Broker
+/// Client (KBC) and KBS, so as to obtain the trusted services or resources of KBS.
+///
+/// # Example
+///
+/// ```rust
+/// use attestation_agent::AttestationAgent;
+/// use attestation_agent::ServiceAPIs;
+///
+/// let mut aa = AttestationAgent::new();
+///
+/// let key_result = aa.decrypt(
+///     "sample_kbc".to_string(),
+///     "https://xxxxx".to_string(),
+///     "example_annotation".to_string()
+/// );
+/// ```
+
+/// `ServiceAPIs` defines the service APIs of attestation agent that need to make requests
+///  to the Relying Party (Key Broker Service) in Confidential Containers.
+///
+/// For every service API, the `kbc_name` and `kbs_uri` is necessary, `kbc_name` tells
+/// attestation agent which KBC module it should use and `kbs_uri` specifics the KBS address.
+pub trait ServiceAPIs {
+    /// `decrypt`is used to decrypt the encrypted information in `annotation`.
+    /// The specific format of `annotation` is defined by different KBC and corresponding KBS.
+    /// The decryption method may be to obtain the key from KBS for decryption, or
+    /// directly send the `annotation` to KBS for decryption, which depends on the
+    /// specific implementation of each KBC module.
+    fn decrypt(&mut self, kbc_name: String, kbs_uri: String, annotation: String)
+        -> Result<Vec<u8>>;
+
+    /// `get_resource` is used to request KBS to obtain confidential resources, including
+    /// confidential data or files.The specific format of `resource_description` is defined by
+    /// different KBC and corresponding KBS.
+    fn get_resource(
```

Comment:
`decrypt` and `get_resource` are slightly inconsistent names. First of all, Rust doesn't really use `get_*` so maybe we should change `get_resource` to something else? `decrypt` is also a little bit vague. What are we decrypting?

Reply:
What about `download_confidential_resource` and `decrypt_image_layer_anno` ?

Reply:
Yeah I think that's good. Might as well say `annotation` instead of `anno`. The name is going to be long either way. You could also just say `decrypt_annotation` or `decrypt_layer_annotation` or something.